fw/http_limits.c

```diff
@@ -738,6 +738,38 @@ frang_http_host_check(const TfwHttpReq *req, FrangAcc *ra)
 	return TFW_PASS;
 }
 
+static int
+frang_http_domain_fronting_check(const TfwHttpReq *req, FrangAcc *ra)
+{
+	TlsCtx *tctx;
+
+	if ((TFW_CONN_TYPE(req->conn) & TFW_FSM_HTTPS) == 0)
+		return TFW_PASS;
+
+	tctx = tfw_tls_context(req->conn);
+
+	if (tctx->vhost != req->vhost) {
+		BasicStr tls_name, req_name;
+		static BasicStr null_name = {"NULL", 4};
+
+		/* Special case of default vhosts */
+		if (req->vhost == NULL
+		    && tfw_vhost_is_default(tctx->vhost))
```

Comment:
`tctx->vhost` can be NULL and `tfw_vhost_is_default()` will crash. @pale-emperor please make sure in https://github.com/tempesta-tech/tempesta-test/issues/212 that there is a test, which requires SNI to match to authority host name, but an actual request doesn't lead to getting a `vhost` in the function.

Reply:
@krizhanovsky what is the semantic meaning of NULL tctx->vhost / req->vhost? I thought that tctx->vhost will always be non-NULL if `TFW_FSM_HTTPS` bit is set in the request connection type.

Reply:
In my understanding, any SNI would make it that tctx->vhost is at least set to the default vhost: https://github.com/tempesta-tech/tempesta/blob/master/fw/tls.c#L830

And I was unpleasantly surprised that TfwHttpReq doesn't follow the same scheme.

Reply:
Probably I didn't get the question. I meant that since you set `ctx->vhost = NULL` in `tfw_tls_sni()`, you need to handle `tcts->vhost == NULL` or `tfw_vhost_is_default(tctx->vhost)` will lead to crash